analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.CSharp8.cs

```diff
@@ -510,3 +510,34 @@ public void TestMethod()
 
     private void Callback() { }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/8149
+public class Repro_8149
+{
+    enum UserType
+    {
+        Internal,
+        External,
+        Other
+    }
+
+    interface IUser
+    {
+        int Id { get; }
+        string LoginName { get; }
+        UserType UserType { get; }
+    }
+
+    static void Check_SwitchExpression(IUser user)
+    {
+        if (user.UserType switch                    // Noncompliant FP, always false
```

Comment:
Personal opinion: after [this poll](https://sonarsource.slack.com/archives/C04P0SB2KJP/p1686055543993659) on the guild of education, I try to stick to this (unwritten) convention.